graph/test/test_srt.c

```diff
@@ -54,15 +54,20 @@ static rage_ConcreteElementType fake_concrete_type = {
     .type = &fake_element_type
 };
 
+#define ERR_MSG_BUFF_SIZE 128
+static char err_msg[ERR_MSG_BUFF_SIZE];
+
 static rage_Error counter_checks(
         sem_t * sync_sem, rage_ElementState * fes, int expected_prep,
         int expected_clean) {
     sem_wait(sync_sem);
     if (fes->prep_counter != expected_prep) {
-        return RAGE_ERROR("Unexpected prep count");
+        snprintf(err_msg, ERR_MSG_BUFF_SIZE, "Unexpected prep count %u != %u", fes->prep_counter, expected_prep);
+        return RAGE_ERROR(err_msg);
     }
     if (fes->clean_counter != expected_clean) {
-        return RAGE_ERROR("Unexpected clean count");
+        snprintf(err_msg, ERR_MSG_BUFF_SIZE, "Unexpected clean count %u != %u", fes->clean_counter, expected_clean);
```

Comment:
I'm not sure whether I like this or not really, the message is nice (and you can get away with it since there's only 1 at a time in the tests), but it's irritatingly hard to factor nicely.